public/app/features/expressions/components/Math.tsx

```diff
@@ -12,24 +14,143 @@ interface Props {
+
+  const toggleShowHelp = useCallback(() => {
+    setShowHelp((show) => !show);
+  }, []);
+
   const onExpressionChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
     onChange({ ...query, expression: event.target.value });
   };
 
+  const styles = useStyles2((theme) => getStyles(theme, showHelp));
+
   return (
-    <InlineField
-      label="Expression"
-      labelWidth={labelWidth}
-      className={css`
-        align-items: baseline;
-      `}
-    >
-      <TextArea value={query.expression} onChange={onExpressionChange} rows={4} placeholder={mathPlaceholder} />
-    </InlineField>
+    <Stack direction="row">
+      <InlineField
+        label="Expression"
+        labelWidth={labelWidth}
+        grow={true}
+        shrink={true}
+        className={css`
+          align-items: flex-start;
+          flex: 0.7;
+        `}
+      >
+        <>
+          <TextArea value={query.expression} onChange={onExpressionChange} rows={5} placeholder={mathPlaceholder} />
+          <Button color="primary" size="sm" onClick={() => toggleShowHelp()}>
+            {showHelp === false ? 'Show' : 'Hide'} help
+          </Button>
+        </>
+      </InlineField>
+      <div className={styles.documentationContainer}>
+        <header className={styles.documentationHeader}>
+          <Icon name="book-open" /> Math operator
+        </header>
+        <div>
+          Run Math operations on one more queries, you reference the query by {'${refId}'} ie. $A, $B, $C etc.
+          <br />
+          Example: <code>$A + $B</code>
+        </div>
+        <header className={styles.documentationHeader}>Available Math functions</header>
+        <div className={styles.documentationFunctions}>
+          <DocumentedFunction
+            name="abs"
+            description="returns the absolute value of its argument which can be a number or a series"
+          />
+          <DocumentedFunction
+            name="is_inf"
+            description="takes a number or a series and returns 1 for Inf values (negative or positive) and 0 for other values"
+          />
+          <DocumentedFunction
+            name="is_nan"
+            description="takes a number or a series and returns 1 for NaN values and 0 for other values"
+          />
+          <DocumentedFunction
+            name="is_null"
+            description="takes a number or a series and returns 1 for null values and 0 for other values"
+          />
+          <DocumentedFunction
+            name="is_number"
+            description="takes a number or a series and returns 1 for all real number values and 0 for non-number"
+          />
+          <DocumentedFunction
+            name="log"
+            description="returns the natural logarithm of of its argument which can be a number or a series"
+          />
+          <DocumentedFunction
+            name="inf, infn, nan, and null"
+            description="The inf, infn, nan, and null functions all return a single value of the name"
```

Comment:
I'm not sure this is clear enough, but I also don't have better suggestions other than let's split it and describe each individually, which I realise is not great. 🤷 

```suggestion
            description="The inf for infinity positive, infn for infinity negative, nan, and null functions all return a single scalar value that matches its name"
```

Reply:
nit